hikari/impl/bot.py

```diff
@@ -23,589 +23,481 @@
         self._intents = intents
-        self._has_aborted = False
-        self._large_threshold = large_threshold
-        self._max_concurrency = 1
-        self._proxy_settings = config.ProxySettings() if proxy_settings is None else proxy_settings
-        self._request_close_event = asyncio.Event()
-        self._rest = rest_client_impl.RESTClientImpl(  # noqa: S106 - Possible hardcoded password
-            connector_factory=self._connector_factory,
-            connector_owner=False,
+        self._proxy_settings = proxy_settings if proxy_settings is not None else config.ProxySettings()
+        self._token = token
+
+        # Caching, chunking, and event subsystems.
+        self._cache: cache.Cache
+        self._chunker: chunker.GuildChunker
+        self._events: event_dispatcher.EventDispatcher
+        events_obj: event_manager_base.EventManagerBase
+
+        if enable_cache:
+            from hikari.impl import stateful_cache
+            from hikari.impl import stateful_event_manager
+            from hikari.impl import stateful_guild_chunker
+
+            cache_obj = stateful_cache.StatefulCacheImpl(self, intents)
+            self._cache = cache_obj
+            self._chunker = stateful_guild_chunker.StatefulGuildChunkerImpl(self, chunking_limit)
+
+            events_obj = stateful_event_manager.StatefulEventManagerImpl(self, cache_obj, intents)
+            self._raw_event_consumer = events_obj.consume_raw_event
+            self._events = events_obj
+        else:
+            from hikari.impl import stateless_cache
+            from hikari.impl import stateless_event_manager
+            from hikari.impl import stateless_guild_chunker
+
+            self._cache = stateless_cache.StatelessCacheImpl()
+            self._chunker = stateless_guild_chunker.StatelessGuildChunkerImpl()
+
+            events_obj = stateless_event_manager.StatelessEventManagerImpl(self, intents)
+            self._raw_event_consumer = events_obj.consume_raw_event
+            self._events = events_obj
+
+        # Entity creation
+        self._entity_factory = entity_factory_impl.EntityFactoryImpl(self)
+
+        # Event creation
+        self._event_factory = event_factory_impl.EventFactoryImpl(self)
+
+        # Voice subsystem
+        self._voice = voice_impl.VoiceComponentImpl(self, self._debug, self._events)
+
+        # RESTful API.
+        self._rest = rest_impl.RESTClientImpl(
             debug=debug,
+            connector_factory=rest_impl.BasicLazyCachedTCPConnectorFactory(),
+            connector_owner=True,
             entity_factory=self._entity_factory,
             executor=self._executor,
             http_settings=self._http_settings,
             proxy_settings=self._proxy_settings,
-            token=token,
-            token_type=constants.BOT_TOKEN,  # nosec
             rest_url=rest_url,
-            version=rest_version,
+            token=token,
+            token_type=constants.BOT_TOKEN,
+            version=6,
```

Comment:
Random thought, would it not make sense for the rest client implementation to be tied to a version rather than having it take a version number that may break it between versions.

Reply:
yep, that's why bot doesn't expose it directly now.